core/src/main/java/io/grpc/internal/DelayedClientTransport.java

```diff
@@ -49,7 +50,8 @@
  * thus the delayed transport stops owning the stream.
  */
 final class DelayedClientTransport implements ManagedClientTransport {
-  private final InternalLogId lodId = InternalLogId.allocate(getClass().getName());
+  // lazily allocated, since it is infrequently used.
+  private final AtomicReference<InternalLogId> logId = new AtomicReference<>();
```

Comment:
DelayedClientTransport is one per ManagedChannelImpl, thus is very scarce. Probably not worth the code complexity to optimize for it.

Reply:
Normally I'd agree with you, but in practice it's annoying that there's gaps in the ID space.  Making the ID's as dense as feasible for Channelz is a better UX.  As is, there are gaps between the IDs when they show up the UI.  The Code is infrequently called, but this is more about allocating IDs more frugally.

Reply:
As soon as any non-channelz object is used, we would start having gaps in the ID space. Channelz IDs are meant to be opaque so this is not a problem IMO.

Reply:
It was a problem for me, a consumer of the API.  I was testing out a channelz application, and when I noticed caps between channels I was making in a for loop, it made me start wondering about where the missing IDs were.   Once the Ids get large enough, the gaps don't matter, but while low, it is nice to have them dense.   This is not a correctness thing, but a nicety.